shared/common-adapters/hoc-timers.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+import React from 'react'
+
+export type TimerFunc = (func: () => void, timing: number) => number
+export type ClearTimerFunc = (id?: ?number) => void
+export type TimerProps = {
+  setTimeout: TimerFunc,
+  clearTimeout: ClearTimerFunc,
+  setInterval: TimerFunc,
+  clearInterval: ClearTimerFunc,
+}
+
+function clearId (clearFunc: (id: number) => void, array: Array<number>, id?: ?number): void {
+  if ((id || id === 0) && array.includes(id)) {
+    array.splice(array.indexOf(id), 1)
+    clearFunc(id)
+  }
+}
+
+const HOCTimers = (ComposedComponent: ReactClass<*>): ReactClass<TimerProps> => class extends React.Component {
```

Comment:
I think the type can be better here.

You want to return a `Class<Component<void, OrignalProps, void>>`. Your input should be type `Class<Component<void, OriginalProps & TimeProps, void>>`.

Where OriginalProps is the component you are wrapping's original props w/o setTimeout, clearTimeout...